langwrappers/src/main/kotlin/org/jetbrains/research/testspark/langwrappers/strategies/JavaKotlinFullTextExtractionStrategy.kt

```diff
@@ -0,0 +1,37 @@
+package org.jetbrains.research.testspark.langwrappers.strategies
+
+import com.intellij.psi.PsiFile
+import org.jetbrains.research.testspark.core.utils.importPattern
+import org.jetbrains.research.testspark.core.utils.packagePattern
+
+/**
+Direct implementor for the Java and Kotlin PsiWrappers
+ */
+class JavaKotlinFullTextExtractionStrategy {
+
```

Comment:
For other languages to adhere the same pattern, the strategy should be made as an interface to be easily substitutable:

```kotlin
interface LanguageTextExtractionStrategy {
   fun extract(file: PsiFile, classText: String): String;
}

class JavaKotlinFullTextExtractionStrategy : LanguageTextExtractionStrategy {
  override fun extract(file: PsiFile, classText: String): String { ... }
}
```

Reply:
I see the method is implemented as static. I cannot say wether it is better or worse, but I would go with a normal method, and then used it as:
```kotlin
class KotlinPsiHelper {
  private val textExtractor = JavaKotlinFullTextExtractionStrategy();
  // ...
  get fullText() = this.textExtractor.extract(...);
}
```
Save for `JavaPsiHelper`.